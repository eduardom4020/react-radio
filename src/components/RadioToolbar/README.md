# RadioToolbar
>This component is a specialization of Toolbar component. It renders a Toolbar with content (title and action buttons).


### Props
These are the props accepted by the component, pay attention to the description, inputs and behaviour:

---

##### title *(string)*
*The title being showed in toolbar. By default it value is 'STATIONS'*

**example**: 
```
<RadioToolbar title='Listen Now!' />
```

---

##### color *(string)*
*The background color in wich the wrapper will be filled. You should pass to it any color code accepted by CSS, like hexadecial, rgb or the color name.*

**example**: 
```
<RadioToolbar color='#ff0000' /> 
```
*this will paint background in red*

---

##### iconStyle *(object)*
*CSS configurations passed to IconButton component.*

**example**: 
```
<RadioToolbar iconStyle={{color: '#ff0000'}} /> 
```

---

##### textStyle *(object)*
*CSS configurations for the header text component.*

**example**: 
```
<RadioToolbar textStyle={{color: '#ff0000'}} /> 
```

---