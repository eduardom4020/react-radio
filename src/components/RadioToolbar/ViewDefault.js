import React from 'react';
import styled from 'styled-components';
import { iconBack, iconSwitch } from '~/assets';
import { Text } from 'react-native';
import Toolbar from '~/src/components/Toolbar';
import IconButton from '~/src/components/IconButton';

const StyledText = styled(Text)`
    color: white;
    font-size: 24px;
    font-weight: bold;
`;

const RadioToolbarViewDefault = props => {
    const { iconStyle, textStyle, color, title } = props;

    return (
        <Toolbar color={color}>
            <IconButton src={iconBack} style={iconStyle} />
            <StyledText style={textStyle}>{title}</StyledText>
            <IconButton src={iconSwitch} style={iconStyle} />
        </Toolbar>
    );
};

export default RadioToolbarViewDefault;