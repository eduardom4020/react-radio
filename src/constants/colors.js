export const BLACK_LIKE = '#20202b';
export const DARK_GREY = '#2e2e3a';
export const MEDIUM_GREY = '#a3acbe';
export const MEDIUM_ORANGE = '#eeae61';
export const METALIC_BLACK = '#191313';