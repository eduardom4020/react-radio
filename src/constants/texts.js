export const APP_TITLE = 'STATIONS';