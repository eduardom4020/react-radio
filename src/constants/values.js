export const DEFAULT_IMAGE_BT_SIZE = 192;
export const DEFAULT_ICON_BT_SIZE = 24;
export const DEFAULT_RADIO_PLAYING = 'CURRENTLY PLAYING'