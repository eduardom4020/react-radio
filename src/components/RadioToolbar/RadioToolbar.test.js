import React from 'react';
import { Text, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { DARK_GREY } from '~/src/constants/colors';
import RadioToolbar from '~/src/components/RadioToolbar';

describe('RadioToolbar', () => {
    test('Render?', () => {
        const component = shallow(
            <RadioToolbar />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom color?', () => {
        const component = shallow(
            <RadioToolbar color={DARK_GREY} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom height?', () => {
        const component = shallow(
            <RadioToolbar height={200} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
    
    test('Render with string title?', () => {
        const component = shallow(
            <RadioToolbar title='Radios App' />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with component title?', () => {
        const component = shallow(
            <RadioToolbar radioTitle={
                <View><Text>Radios Toolbar</Text></View>
            } />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
});