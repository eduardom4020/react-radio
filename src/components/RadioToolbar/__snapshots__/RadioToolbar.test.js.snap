// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioToolbar Render with component title? 1`] = `
<Styled(View)
  radioTitle={
    <View>
      <Text>
        Radios Toolbar
      </Text>
    </View>
  }
>
  <IconButtonViewDefault
    src={
      Object {
        "testUri": "../../../assets/back-arrow.png",
      }
    }
  />
  <Styled(Text) />
  <IconButtonViewDefault
    src={
      Object {
        "testUri": "../../../assets/switch.png",
      }
    }
  />
</Styled(View)>
`;

exports[`RadioToolbar Render with custom color? 1`] = `
<Styled(View)
  color="#2e2e3a"
>
  <IconButtonViewDefault
    src={
      Object {
        "testUri": "../../../assets/back-arrow.png",
      }
    }
  />
  <Styled(Text) />
  <IconButtonViewDefault
    src={
      Object {
        "testUri": "../../../assets/switch.png",
      }
    }
  />
</Styled(View)>
`;

exports[`RadioToolbar Render with custom height? 1`] = `
<Styled(View)
  height={200}
>
  <IconButtonViewDefault
    src={
      Object {
        "testUri": "../../../assets/back-arrow.png",
      }
    }
  />
  <Styled(Text) />
  <IconButtonViewDefault
    src={
      Object {
        "testUri": "../../../assets/switch.png",
      }
    }
  />
</Styled(View)>
`;

exports[`RadioToolbar Render with string title? 1`] = `
<Styled(View)>
  <IconButtonViewDefault
    src={
      Object {
        "testUri": "../../../assets/back-arrow.png",
      }
    }
  />
  <Styled(Text)>
    Radios App
  </Styled(Text)>
  <IconButtonViewDefault
    src={
      Object {
        "testUri": "../../../assets/switch.png",
      }
    }
  />
</Styled(View)>
`;

exports[`RadioToolbar Render? 1`] = `
<Styled(View)>
  <IconButtonViewDefault
    src={
      Object {
        "testUri": "../../../assets/back-arrow.png",
      }
    }
  />
  <Styled(Text) />
  <IconButtonViewDefault
    src={
      Object {
        "testUri": "../../../assets/switch.png",
      }
    }
  />
</Styled(View)>
`;
